src/setuptools_scm/git.py

```diff
@@ -26,9 +27,18 @@ def do_ex(self, cmd):
 
     @classmethod
     def from_potential_worktree(cls, wd):
-        real_wd, _, ret = do_ex("git rev-parse --show-toplevel", wd)
+        wd = os.path.abspath(wd)
+        real_wd, _, ret = do_ex("git rev-parse --show-prefix", wd)
+        real_wd = real_wd[:-1]  # remove the trailing pathsep
         if ret:
             return
+        if not real_wd:
+            real_wd = wd
+        else:
+            assert wd.replace("\\", "/").endswith(real_wd)
```

Comment:
for nitpicking maybe lets add a comment that the length is not changed by the replace,
as the escape always trips me up, alternatively maybe a raw string could also help

also great job, this is merge-able as is, do you want to squish some commits?

Reply:
> alternatively maybe a raw string could also help

ok then I will use raw strings

> also great job, this is merge-able as is, do you want to squish some commits?

maybe squash while merge?

Reply:
no raw string doesn't help, because it raises `SyntaxError: EOL while scanning string literal`

Reply:
let me add a comment.

Reply:
i  can do that if you are fine with it (my personal stance is that  i only change history created by others if they request it or consent to it )

i'll enable squash merges for this repo now ^^

Reply:
done https://github.com/pypa/setuptools_scm/pull/520/commits/92c5422207df8b376232ace6d311575d75feb415

Reply:
> i can do that if you are fine with it (my personal stance is that i only change history created by others if they request it or consent to it )

It's ok I don't mind, it could some problems if squash it myself, rather than GitHub squashing it.